dev/merge_spark_pr.py

```diff
@@ -135,11 +135,12 @@ def merge_pr(pr_num, target_ref, title, body, pr_repo_desc):
         continue_maybe(msg)
         had_conflicts = True
 
+    # First commit author should be considered as the primary author when the rank is the same
     commit_authors = run_cmd(
-        ["git", "log", "HEAD..%s" % pr_branch_name, "--pretty=format:%an <%ae>"]
+        ["git", "log", "HEAD..%s" % pr_branch_name, "--pretty=format:%an <%ae>", "--reverse"]
     ).split("\n")
     distinct_authors = sorted(
-        set(commit_authors), key=lambda x: commit_authors.count(x), reverse=True
+        list(dict.fromkeys(commit_authors)), key=lambda x: commit_authors.count(x), reverse=True
```

Comment:
This generated an ordered set.

Reply:
There is a `sorted()` method out there. Do we still need it?

Reply:
Actually the sort is with `key=lambda x: commit_authors.count(x)`. For the same elements, I think it keeps its natural order.  IIRC `sorted` in Python is Timesort that is a stable sort that keeps the same order of same keys

Reply:
Yes, please see the following, @gengliangwang .
```python
>>> commit_authors=['a', 'b', 'c']
>>> sorted(set(commit_authors), key=lambda x: commit_authors.count(x), reverse=True)
['c', 'a', 'b']
>>> sorted(list(dict.fromkeys(commit_authors)), key=lambda x: commit_authors.count(x), reverse=True)
['a', 'b', 'c']
```

Reply:
`sorted` preserves the order by itself. The root cause of the problem was `set(...)`.

Reply:
And, the fact that `git log`'s default order is a reverse order.

Reply:
So, we need both, `--reverse` and `list(dict.fromkeys(...))`.

Reply:
I am seeing
```
>>> commit_authors=['a', 'b', 'c', 'c']
>>> sorted(list(dict.fromkeys(commit_authors)), key=lambda x: commit_authors.count(x), reverse=True)
['c', 'a', 'b']
```

Reply:
When a person does more than two commits, that guy is the main author. I think this Pr only fixes when the number of commits from each author is same. (Note that we sort by the number of commits, not the author's name)

Reply:
Oh, I see. I thought the proposal was to make the author of first commit as the PR author. My bad.

Reply:
Yes, I thought that idea too, but decided to fix the tie only in this PR, @gengliangwang .
> I thought the proposal was to make the author of first commit as the PR author.